### PR TITLE
ARROW-18118: [Release][Dev] Fix problems in 02-source.sh/03-binary-submit.sh for 10.0.0-rc0

### PR DESCRIPTION
**Pull request by kou**
SOURCE_PR in dev/release/02-source.sh didn't work because it refers undefined variable (`rc_number` is referred not `rc`). This change includes the following changes including the fix of the undefined variable problem:

- Fix wrong variable use. (Variable name is also changed to adjust other variable names: `release_candidate` -> `rc`)
- The ARROW_GITHUB_API_TOKEN environment variable is used for GitHub Personal Access Token but the CROSSBOW_GITHUB_TOKEN environment variable is used in others. So this also uses CROSSBOW_GITHUB_TOKEN. (Documentation is also updated. This change also includes unrelated minor updates. Sorry.)

dev/release/03-binary-submit.sh didn't work because:

- It refers undefined variable (`release_candidate_branch`).
- `archery crossbow report-pr` searches only `base=master` pull requests. (A verify pull request uses `base=maint-X.Y.Z`.)

This change fixes them. This change also adds `--no-fetch` to `archery crossbow latest-prefix` like other command lines to reduce execution time.

### CODE REVIEW

**Comment by kou** [comment]
@raulcd What do you think about this?

File: dev/release/03-binary-submit.sh [hunk 1]
```diff
@@ -29,6 +29,7 @@ rc_number=$2
 version_with_rc="${version}-rc${rc_number}"
```
Comment (raulcd) [inline dev/release/03-binary-submit.sh:29]: Sorry about that, I think this is coming from moving the command between steps 03 to 02 when I was doing it initially. Should we change the `rc_number` var name here to `rc` to be more consistent between scripts?

**Review by raulcd — APPROVED** [review]
Thanks @kou ! Minor nit feel free to ignore, thanks for fixing that!

File: dev/release/03-binary-submit.sh (hunk 1 above)
Comment (kou) [reply dev/release/03-binary-submit.sh]: OK. I've changed to use `rc` here.

**Pull request merged**